### PR TITLE
holdings: fix electronic holding availability

### PR DESCRIPTION
**Pull request by zannkukai**
* Mark an electronic holding ass always available.
* Closes rero/rero-ils#1872.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/holdings/api.py [hunk 1]
```diff
@@ -271,13 +269,11 @@ def get_note(self, note_type):
...
-        """."""
-        for item in items:
-            if item.available:
-                return True
-        return False
+    def _exists_available_child(self):
+        """Check if at least one child of this holding is available."""
+        return exists_available_item(
+            Item.get_items_pid_by_holding_pid(self.pid)
+        )
```
Comment (sebastiendeleze) [inline rero_ils/modules/holdings/api.py:276]: Better to group same functions (property, static, instance) together.

File: rero_ils/modules/holdings/models.py [hunk 2]
```diff
@@ -42,6 +42,14 @@ class HoldingMetadata(db.Model, RecordMetadataBase):
     __tablename__ = 'holding_metadata'
 
 
+class HoldingTypes:
```
Comment (sebastiendeleze) [inline rero_ils/modules/holdings/models.py:45]: Maybe can be extended from `Enum`

File: rero_ils/modules/items/api/record.py [hunk 3]
```diff
@@ -218,7 +219,7 @@ def link_item_to_holding(cls, record, mode):
 
         if (
             mode == 'create' and record.get('holding')) or (
-            old_holding_type in ['serial', 'electronic']
+            old_holding_type in [HoldingTypes.SERIAL, HoldingTypes.ELECTRONIC]
```
Comment (sebastiendeleze) [inline rero_ils/modules/items/api/record.py:222]: `Enum` class can be iterated.

File: rero_ils/modules/items/utils.py [hunk 4]
```diff
@@ -48,3 +48,22 @@ def item_location_retriever(item_pid):
...
+
+def exists_available_item(items=[]):
+    """Check if at least one item of the list are available.
+
+    Caution: `items` param couldn't be a generator otherwise the
+    :param items: the items to check. Either as a list of item pids, either as
+                  a list of item resources.
+    :return True if one item is available; false otherwise.
+    """
+    from rero_ils.modules.items.api import Item
```
Comment (sebastiendeleze) [inline rero_ils/modules/items/utils.py:61]: This import can be moved at the top of the file?

**Review by sebastiendeleze — APPROVED** [review]
Type in commit message `ass`

File: rero_ils/modules/items/utils.py (hunk 4 above)
Comment (zannkukai) [reply rero_ils/modules/items/utils.py:61]: nope : Circular import :'(

File: rero_ils/modules/items/api/record.py (hunk 3 above)
Comment (zannkukai) [reply rero_ils/modules/items/api/record.py:222]: I can't enumerate on the Enum class itself because it contains 3 values ; Only 2 are valid in this test.

**Pull request closed without merging**